src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/TransmissionStorage.cs

```diff
@@ -139,24 +140,34 @@ public virtual Transmission Dequeue()
                         return transmission;
                     }
                 }
-                catch (UnauthorizedAccessException)
+                catch (UnauthorizedAccessException uae)
                 {
                     if (file == null)
                     {
-                        return null; // because the process does not have permission to access the folder
+                        TelemetryChannelEventSource.Log.TransmissionStorageDequeueUnauthorizedAccessException(this.folder.Name, uae.ToString());
+                        return null; // Because the process does not have permission to access the folder.
                     }
 
-                    if (lastInaccessibleFileName != file.Name)
+                    string name = file.Name;
+                    if (this.badFiles.TryAdd(name, null))
+                    {
+                        TelemetryChannelEventSource.Log.TransmissionStorageInaccessibleFile(name);
+                    }
+                    else
                     {
-                        lastInaccessibleFileName = file.Name;
-                        continue; // because another thread is loading this file right now
+                        // The same file has been inaccessible more than once because the process does not have permission to modify this file.
+                        TelemetryChannelEventSource.Log.TransmissionStorageUnexpectedRetryOfBadFile(name);
                     }
 
-                    throw; // because the process does not have permission to modify this file
+                    Thread.Sleep(random.Next(1, 100)); // Sleep for random time of 1 to 100 milliseconds to try to avoid future timing conflicts.
```

Comment:
I don't believe so because eventually we will run out of files which can be processed and then exit as normal.  The only change in the logic here is to stop throwing what appears to be an unhandled exception that didn't address the underlying issue.